### DISCUSSION TITLE
Exception assertions and 'And'

### DISCUSSION THREAD

**Discussion by JesseNicholsonStorable** [fluentassertions/fluentassertions#1946] [Q&A] [answered]
I am upgrading our FA version from 5.10.3 to 6.7.0 and I am concerned that a number of our current test are using a no-longer used feature.  We have many tests that look like 
```cs
sut.Awaiting(x => x.Handle(command, CancellationToken.None))
    .Should()
    .ThrowAsync<RequestFailureException>()
    .And.Error.Code.Should()
    .Be(StandardErrorCodes.ResourceNotFound.Code);
```

Originally they were breaking on the throw, which was an easy enough fix, however now they all break on the `And` that follows immediately after.  

I have tried searching for a topic like this but a search for why `And` isn't working doesn't seem to achieve much.

**Comment by dennisdoomen** [comment, marked as answer]
Because using `Wait` and `Result` in `async` code is widely discouraged, in v6 we have been very careful to remove any blocking code. You can do a couple of things to workaround this:
```cs
await sut.Awaiting(x => x.Handle(command, CancellationToken.None))
    .Should()
    .ThrowAsync<RequestFailureException>()
    .Where(e => e.Error.Code == StandardErrorCodes.ResourceNotFound.Code);
```
or
```cs
(await sut.Awaiting(x => x.Handle(command, CancellationToken.None))
    .Should()
    .ThrowAsync<RequestFailureException>())
    .And.Error.Code.Should()
    .Be(StandardErrorCodes.ResourceNotFound.Code);
```
PS: `Awaiting` returns a `Task`, so you must using `await` as well.

**Reply by JesseNicholsonStorable** [reply]
Thank you, that helps a lot!
